src/components/Sidebars/Payback.js

```diff
@@ -19,11 +67,70 @@ const Payback = ({ cdp, reset }) => {
   const [liquidationPrice, setLiquidationPrice] = useState(0);
   const [collateralizationRatio, setCollateralizationRatio] = useState(0);
 
+  const [proxyAddress, setProxyAddress] = useState(null);
+
+  const [
+    { allowanceStatus, displayAllowanceToggle, allowanceTxProcessing },
+    dispatch
+  ] = useReducer(reducer, initialState);
+
+  useEffect(() => {
+    (async () => {
+      setProxyAddress(await maker.service('proxy').ensureProxy());
+      await determineAllowanceStatus();
+    })();
+  }, [proxyAddress]);
+
+  const checkAllowance = async () => {
+    const daiToken = maker.getToken('MDAI');
+    return (await daiToken.allowance(maker.currentAddress(), proxyAddress)).eq(
+      MAX_UINT_BN
+    );
+  };
+
+  const determineAllowanceStatus = async () => {
+    if (proxyAddress) {
```

Comment:
under what conditions would we not have the `proxyAddress` here?

Reply:
Yes, this is an issue that I kept running into when trying to set the `proxyAddress` in the first case. As we have the `proxyAddress` as a dependency in the `useEffect()` it will be called twice, the first time on component mount and then the second time when `proxyAddress` is set [here](https://github.com/makerdao/mcd-cdp-portal/pull/57/files/ad547e471387e08a1408d6061e9efe5e4581d049#diff-218015994f32dbcc7fd976a9262630a8R79). 

The issue this `if` is solving is that when I was initially calling the `useEffect()` with an empty dependency array, and so only being called on component mount. What then occurred was that the `setProxyAddress()` would not have the `proxyAddress` set by the time `determineAllowanceStatus()` was called. Having the `useEffect()` to effectively loop twice, setting the proxy address first and then doing the allowance lookup works but I feel like its a bit of a bandaid solution.

I'll dig into how the setState hooks work in regards to the js event loop, as it would make sense that `determineAllowanceStatus()` is being called ahead of `setProxyAddress()`